### PR TITLE
feat: anmodning og download - engros uden basisdata

### PR DESCRIPTION
**Pull request by michaelmajgaard**
# Description

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [x] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Generators/ChargeLinkPeriodsFileGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        var csvHelper = new CsvWriter(destination, new CultureInfo(filter.CsvFormatLocale ?? "en-US"));
+        csvHelper.Context.RegisterClassMap<SettlementReportChargeLinkPeriodsResultRowMap>();
+
+        await using (csvHelper.ConfigureAwait(false))
+        {
+            csvHelper.WriteHeader<SettlementReportChargeLinkPeriodsResultRow>();
+            await csvHelper.NextRecordAsync().ConfigureAwait(false);
+
+            // TODO: Fix data source.
```
Comment (michaelmajgaard) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Generators/ChargeLinkPeriodsFileGenerator.cs:43]: Should be removed?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/SettlementReportWholesaleResultQueries.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        var statement = new SettlementReportWholesaleResultCountQueryStatement(_deltaTableOptions, filter);
+
+        await foreach (var nextRow in _databricksSqlWarehouseQueryExecutor.ExecuteStatementAsync(statement, Format.JsonArray).ConfigureAwait(false))
+        {
+            var rawValue = new DatabricksSqlRow(nextRow)[SettlementReportWholesaleResultCountQueryStatement.Columns.Count];
+            return SqlResultValueConverters.ToInt(rawValue)!.Value;
+        }
+
+        return 0;
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/SettlementReportWholesaleResultQueries.cs:53]: Throw exception here?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/SettlementReportEnergyResultQueries.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    private async Task<int> InternalCountAsync(DatabricksStatement statement)
+    {
+        await foreach (var nextRow in _databricksSqlWarehouseQueryExecutor.ExecuteStatementAsync(statement, Format.JsonArray).ConfigureAwait(false))
+        {
+            var rawValue = new DatabricksSqlRow(nextRow)[SettlementReportEnergyResultCountColumns.Count];
+            return SqlResultValueConverters.ToInt(rawValue)!.Value;
+        }
+
+        return 0;
```
Comment (defectiveAi) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/SettlementReportEnergyResultQueries.cs:89]: Throw an exception?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Generators/ChargeLinkPeriodsFileGenerator.cs (hunk 1 above)
Comment (defectiveAi) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SettlementReports_v2/Generators/ChargeLinkPeriodsFileGenerator.cs:43]: Probably

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**